made compatible with Rider 2024.1.5

abieganski opened the pull request:
Hi @DeagleGross, I think I managed to make it compatible with the latest Rider (2024.1.5).

110 files changed, I know... But I had to sort of start from scratch -- get the latest Rider plugin template and copy stuff from your code over into it.

abieganski committed:
made compatible with Rider 2024.1.5

DeagleGross commented:
Hey,
thanks for contribution.

Why 2024.1.5 is not compatible with an existing code base and you had to re-do it completely? Before it was relatively easy to support new versions: i.e. https://github.com/DeagleGross/SharpCoachPlugin/pull/27

Also at least - why did you delete a ci pipeline? How can we make sure tests work? :)

abieganski committed:
bring back Tests

abieganski commented:
Yeah, so I tried to just change the version number and upgrade the nuget packages -- and that's when a nuget package dll hell broke loose. I could not get the packages to not conflict.

>Also at least - why did you delete a ci pipeline? How can we make sure tests work? :)

Ooops, this is my first time dealing with a Rider plugin, I knew I'd mess something up.
I put all the tests back in now.

abieganski committed:
bring back sample projects

abieganski committed:
fixed plugin.xml

abieganski committed:
brought back MapModelsAction.kt and CoachSharpOptionsPage.kt

abieganski committed:
renamed Resources (removed "SampleContext")

abieganski committed:
brought back .github

abieganski committed:
brought back docs

abieganski committed:
brought back SharpCoachPluginModel.kt

abieganski committed:
lower case RiderPluginId

abieganski committed:
brought back photos

abieganski committed:
replaced InBorder with WithLineBorder

abieganski committed:
cleaned up MapModelsAction

abieganski commented:
OK, I brought back lots of stuff that I missed when copying over.
I think everything should be there now.

abieganski commented:
Hi @DeagleGross, it looks like all the tests have passed. Would you like to publish this version?

DeagleGross commented:
I don't see any tests being run. Log of CI just says build happened. Maybe i miss something

abieganski committed:
made compatible with Rider 2024.1.5 - minimal changes vs the previous version

abieganski commented:
Hi @DeagleGross, me again :)

So I changed tack: reverted to the latest main, and then made the necessary changes in build.gradle etc.

Afterward I had to reference newer versions of the JetBrains nuget packages -- I changed the Nuget package references manually in the csproj files, and was able to get rid of the version conflicts.

Finally, I ran the tests locally and had to make some small adjustments in the *.gold files.

If you could approve the build/CI workflow now, we should be able to see if everything works.

Thanks!
